docs/pages/database-access/auto-user-provisioning/mariadb.mdx

```diff
@@ -51,6 +51,7 @@ CREATE USER 'teleport-admin' IDENTIFIED WITH AWSAuthenticationPlugin AS 'RDS';
 GRANT PROCESS, CREATE USER ON *.* TO 'teleport-admin';
 GRANT SELECT ON mysql.roles_mapping TO 'teleport-admin';
 GRANT UPDATE ON mysql.* TO 'teleport-admin'; -- For SET DEFAULT ROLE FOR
+GRANT SELECT ON *.* TO 'teleport-admin'; -- For checking if users own resources before dropping them.
```

Comment:
Could we mark this permission as **optional** here and add a note somewhere in the doc explaining how this resources-checking works in drop mode?

Reply:
I think we're lacking a proper definition of what `BEST_EFFORT_DROP` is supposed to do.

The definition I used to implement it for MySQL/PostgreSQL/Redshift/MariaDB was: If there won't be any resource disruption, drop the user otherwise, disable it.

Given this, users must give the `teleport-admin` this permission if they are willing to use this mode. We can assume the least permission here, but if users change their mode without updating the admin permissions, this will cause the procedure to malfunction, and we cannot detect that and raise a warning. So, it might be better for UX to ensure this is set up from the beginning.

What do you think about this definition? If we agree on that (or any other), I don't think we need to explain it over every engine, just update the mode description. cc @smallinsky

Reply:
For `optional`, what I meant is that it's not required for `keep` mode. I guess alternatively we can just say something like "only required for best_effort_drop mode" here? (Agreed that we shouldn't tell user to omit these for drop mode even though that situation is really a "secret" force drop mode)

> I think we're lacking a proper definition of what BEST_EFFORT_DROP is supposed to do.
>
Agreed, it would be great if we could expand the current definition. 

>  If there won't be any resource disruption, drop the user. Otherwise, disable it.

"resource disruption" is not very clear to me. Maybe "If no resources depend on the user, drop the user". @ptgott is probably the better person for wording.

Reply:
> If no resources depend on the user, drop the user

This is fine, too. Since we're dropping users, we can be specific here.

Reply:
and please update both tabs. thanks!